packages/server/lib/makeDataContext.ts

```diff
@@ -46,9 +47,20 @@ export function makeDataContext (options: MakeDataContextOptions): DataContext {
       close: browsers.close,
       getBrowsers,
       async ensureAndGetByNameOrPath (nameOrPath: string) {
+        const projectConfig: FullConfig = await (ctx as DataContext).project.getConfig()
+        const declaredBrowsers = projectConfig.browsers?.map<FoundBrowser | undefined>((b) => {
+          if (!b.path) return
+
+          return {
+            path: b.path,
+            version: b.version,
+            majorVersion: b.majorVersion,
+            custom: true,
+          }
+        }).filter<FoundBrowser>((b): b is FoundBrowser => b != null) || []
         const browsers = await ctx.browser.machineBrowsers()
 
-        return await ensureAndGetByNameOrPath(nameOrPath, false, browsers)
+        return await ensureAndGetByNameOrPath(nameOrPath, false, [...declaredBrowsers, ...browsers])
```

Comment:
I wonder if we want to do a check to make sure there isn't any duplicates here - it shouldn't really happen, but if someone adds a default browser via `cypress.config.ts` for whatever reason, it'd show up twice.

Reply:
What would you de-dupe on? family / major version / minor version / path?

Reply:
Just ran across a `removeDuplicates` method elsewhere which uses name and version
```
  const removeDuplicates = (val) => {
    return _.uniqBy(val, (browser: FoundBrowser) => {
      return `${browser.name}-${browser.version}`
    })
  }
```

Reply:
If this is the pattern, maybe we just go with that. I wonder if we can make that a reused utility? Thoughts?

Reply:
I've consolidated this into `BrowserDataSource` and exported it from there.